src/Pact/Native/Internal.hs

```diff
@@ -198,6 +204,26 @@ enforceGuardDef dn =
       _ -> argsError i as
 
 
+enforceVerifierDef :: NativeDef
+enforceVerifierDef = defRNative
+  "enforce-verifier"
+  enforceVerifier
+  (funType tTyBool [("verifiername", tTyString)])
+  [ LitExample $ "(enforce-verifier 'COOLZK)"
+  ]
+  "Enforce that a verifier is in scope."
+  where
+  enforceVerifier :: RNativeFun e
+  enforceVerifier i as = case as of
+    [TLitString verifierName, TList {_tList = (traverse (preview _TLitString) -> Just args) }] -> do
```

Comment:
I don't think this is the right signature. Requiring pact code to re-specify the args to the verifier is simply too verbose. I appreciate that this means you can only use a given verifier once, but the cost of having to reference a large opaque blob in pact code is too high. (Also funtype above is wrong but maybe it's ... right :) )

Reply:
Reverted to just the string argument.